charmhelpers/contrib/storage/linux/ceph.py

```diff
@@ -856,19 +856,35 @@ def _keyring_path(service):
     return KEYRING.format(service)
 
 
-def create_keyring(service, key):
-    """Create a new Ceph keyring containing key."""
+def add_key(service, key):
+    """
+    Add a key to a keyring.
+
+    Creates the keyring if it doesn't already exist.
+
+    Logs and returns if the key is already in the keyring.
+    """
     keyring = _keyring_path(service)
     if os.path.exists(keyring):
-        log('Ceph keyring exists at %s.' % keyring, level=WARNING)
-        return
+        with open(keyring) as ring:
+            if key not in ring.read():
```

Comment:
Also since ware nit picking: It is clearer to do

if key in ring.read():
  log(" ...
  return

log("Adding key ....
cmd = ...

Reply:
Done.